examples/nlp/pretraining_BERT.py

```diff
@@ -0,0 +1,481 @@
+have been showing that a similar technique can be useful in many natural language tasks.
+
+BERT makes use of Transformer, an attention mechanism that learns contextual relations
+between words (or sub-words) in a text. In its vanilla form, Transformer includes two
+separate mechanisms — an encoder that reads the text input and a decoder that produces
+a prediction for the task. Since BERT’s goal is to generate a language model, only the
+encoder mechanism is necessary. The detailed workings of Transformer are described in
+a paper by Google.
+
+As opposed to directional models, which read the text input sequentially
+(left-to-right or right-to-left), the Transformer encoder reads the entire
+sequence of words at once. Therefore it is considered bidirectional, though
+it would be more accurate to say that it’s non-directional. This characteristic
+allows the model to learn the context of a word based on all of its surroundings
+(left and right of the word).
+
+When training language models, there is a challenge of defining a prediction goal.
+Many models predict the next word in a sequence (e.g. The child came home from _ ),
+a directional approach which inherently limits context learning. To overcome this
+challenge, BERT uses two training strategies:
+
+### Masked Language Modeling (MLM)
+Before feeding word sequences into BERT, 15% of the words in each sequence are replaced
+with a `[MASK]` token. The model then attempts to predict the original value of the masked
+words, based on the context provided by the other, non-masked, words in the sequence.
+
+### Next Sentence Prediction (NSP)
+In the BERT training process, the model receives pairs of sentences as input and learns to
+predict if the second sentence in the pair is the subsequent sentence in the original
+document. During training, 50% of the inputs are a pair in which the second sentence is the
+subsequent sentence in the original document, while in the other 50% a random sentence
+from the corpus is chosen as the second sentence. The assumption is that the random sentence
+will be disconnected from the first sentence.
+
+Though Google provides a pre-trained checkpoint of BERT on English, you might often need
+to either pre-train the model from scratch for a different language, or do a
+continued-pretraining to fit the model to your domain. In this notebook, we will pre-train
+BERT from scratch optimizing both MLM and NSP objectves using 🤗 Transformers on the `WikiText`
+English dataset loaded from 🤗 Datasets.
+"""
+
+"""
+## Setup
+"""
+
+"""
+### Installing the requirements
+"""
+
+"""shell
+pip install git+https://github.com/huggingface/transformers.git
+pip install datasets
+pip install huggingface-hub
+pip install nltk
+"""
+
+"""
+### Importing the necessary libraries
+"""
+
+import nltk
+import random
+
+import tensorflow as tf
+from tensorflow import keras
+
+nltk.download("punkt")
+# Only log error messages
+tf.compat.v1.logging.set_verbosity(tf.compat.v1.logging.ERROR)
```

Comment:
Do not use `compat.v1` for any reason. It is legacy and subject to future removal.